pyfar/dsp/window.py

```diff
@@ -0,0 +1,378 @@
+                    W[i_A] = (-1)**m[i_A] * np.cosh(M* np.arccosh(A[i_A]))
+                else:
+                    W[i_A] = (-1)**m[i_A] * np.cos(M* np.arccos(A[i_A]))
+                    
+            w = np.real(np.fft.ifft(W))
+            w[0] = w[0]/2
+            # w = np.append(w, w[0])
+            w = w/np.max(w)
+        else: # symmetric length
+            M = (length-1)
+            gamma = sideLobeLevel/20
+            alpha = np.cosh(np.arccosh(10**gamma)/M)
+            m = np.arange(M)
+            
+            A = np.abs(alpha * np.cos(np.pi*m/M))
+            
+            W = np.zeros(np.size(A))
+            for i_A in range(np.size(A)):
+                if A[i_A] > 1:
+                    W[i_A] = (-1)**m[i_A] * np.cosh(M* np.arccosh(A[i_A]))
+                else:
+                    W[i_A] = (-1)**m[i_A] * np.cos(M* np.arccos(A[i_A]))
+                    
+            w = np.real(np.fft.ifft(W))
+            w[0] = w[0]/2
+            w = np.append(w, w[0])
+            w = w/np.max(w)
+        return w
+    else:
+        raise ValueError('window length has to be >=0 !')
+        
+
+def window(signal, windowType, windowLength, windowStartIndex=0):
+    """windows a  pyfar signal with selected window type and length
+
+    Parameters
+    ----------
+    signal : Signal object
+        An audio signal object from the pyfar signal class
+    
+    windowType: string
+        select window type from 'rect', 'hann', 'hamming', 'blackman', 
+        'bartlett', 'kaiser', 'kaiserBessel' , 'flattop' or 'dolphChebychev'
+    
+    windowLength: int
+        sets window length in samples of selected window
+        
+    windowStartIndex: int
+        sets the start index! of the window
+        
+    Returns
+    -------
+    out : pyfar
+        window coefficients
+    """
+    
+    #check input and default values
+    if not isinstance(signal, Signal):
+        raise TypeError('Input data has to be of type: Signal.')
+        
+    if not isinstance(windowType, str):
+        raise TypeError('''windowType has to be a string of the following:
+                        'rect', 'hann', 'hamming', 'blackman', 'bartlett', 'kaiser',
+                        'kaiserBessel' , 'flattop' or 'dolphChebychev'.''')
+    if not isinstance(windowLength,int):
+        raise TypeError('windowLength has to be of type int.')
+        
+    if not isinstance(windowStartIndex,int) or windowStartIndex < 0:
+        raise TypeError('windowStartIndex has to be of type int and >= 0 .')
+    
+    #create selected window with simulated switch case   
+    switcher = {
+        'rect':rect,
+        'hann':hann,
+        'hamming':hamming,
+        'blackman':blackman,
+        'bartlett':bartlett,
+        'kaiser':kaiser,
+        'kaiserBessel':kaiserBessel,
+        'flattop':flattop,
+        'dolphChebychev':dolphChebychev,
+        }
+    if windowType in switcher:
+        win = switcher.get(windowType)(windowLength)
```

Comment:
This might be more common:
`win = switcher[windowType](windowLength)`

and you can pipe through the kwargs with
`win = switcher[windowType](windowLength, **kwargs)`

`window` might be a more verbose variable name